### ISSUE TITLE
[oAuth] doesn't work with subpath: end_of_popup_response.js get 404

### ISSUE THREAD

**Issue by Buom01** [meteor/meteor#6707]
Hi,
When I'm logging in with an external service, all works except a little detail that block all, I'm using a subpath;

> Failed to load resource: the server responded with a status of 404 (Not Found)

Because it's try to get `end_of_popup_response.js` without subpath

So with this bug, i'm not logged in, and the popup is keeped open and white

Packages: 
- Using facebook, google, and twitter login package

Context:
- Tried with my deployed app :/ with Firefox and Chromium on Linux with a logged user (that should works)
- Meteor 1.3 on linux, using demeteorize

Reproduce:
- https://github.com/Buom01/meteor-oauth-bug
  You will need to create github keys

**Comment by Buom01** [comment]
While waiting a patch:
```bash
$ meteor add meteorhacks:picker
```
```javascript
Picker.route('/packages/oauth/end_of_popup_response.js', function(params, req, res, next) {
  res.writeHead(302, {
    'Location': "/y/packages/oauth/end_of_popup_response.js"
  });
  res.end();
});
```

**Comment by abernix** [comment]
@Buom01 I know this is a relatively old issue, but I think another related issue just came in for it.  See my comments here: https://github.com/meteor/meteor/issues/7979#issuecomment-256851071  This sub-path configuration is likely not very common and therefore this hasn't got much attention, but your help in identifying it and seeing if we can work out a fix for it would be appreciated.

Maybe pop into that other issue and see if it sounds like the same problem to you?  I made a recommendation on how it might possibly be fixed but I'm curious what you think.

**Comment by keyscores** [comment]
@abernix I've noticed that OAuth has no tests.  Am I mistaken or is this the case?
See here:
https://github.com/meteor/meteor/blob/master/packages/accounts-oauth/oauth_tests.js

**Comment by abernix** [comment]
@keyscores The tests which exist are in the parent packages: [`oauth`](https://github.com/meteor/meteor/blob/devel/packages/oauth/oauth_tests.js#L16), [`oauth2`](https://github.com/meteor/meteor/blob/devel/packages/oauth2/oauth2_tests.js#L1) and [`oauth1`](https://github.com/meteor/meteor/blob/devel/packages/oauth1/oauth1_tests.js#L1).

**Comment by keyscores** [comment]
@abernix thanks for the quick response. Can you quickly point me to the code which handles the route callback request: http://domain.com/_oauth/github?close 

Off topic: I'd like to set up a method which does not create a new user and just appends the service to a current user.

**Comment by abernix** [comment]
@keyscores I believe the entry point for requests to that URL would be handled in [`oauth2_server.js`](https://github.com/meteor/meteor/blob/devel/packages/oauth2/oauth2_server.js#L2)  or [`oauth1_server.js`](https://github.com/meteor/meteor/blob/devel/packages/oauth1/oauth1_server.js#L25) (depending on the oAuth version used by the service) in the `requestHandlers` functions.

In regard to your "off-topic" part, the `Meteor.users` insert/update is ultimately handled by [this code](https://github.com/meteor/meteor/blob/devel/packages/accounts-base/accounts_server.js#L1395) in `accounts-base`, but called [here in `accounts-oauth`](https://github.com/meteor/meteor/blob/devel/packages/accounts-oauth/oauth_server.js#L55). Maybe make a forum post if you need further guidance as I'm not sure any of this is exactly "on-topic" for this particular issue. Hope this helps! 😄

**Comment by abernix** [comment]
In regard to the original issue reported here:

@Buom01 Thank you for your easy and well-explained reproduction steps – and for the work-around.  Based on that repro, I'm going to mark this as a confirmed bug and copy my comments from #7979 (a duplicate issue) which I believe explains the problem (and possible solution) here in case someone wants to address it:

> it's possible that the [`end_of_popup_response.html` template](https://github.com/meteor/meteor/blob/devel/packages/oauth/end_of_popup_response.html#L9) needs treatment similar to the [boilerplate HTML](https://github.com/meteor/meteor/blob/devel/packages/boilerplate-generator/boilerplate_web.browser.html#L24-L25) (which is [set here](https://github.com/meteor/meteor/blob/devel/packages/webapp/webapp_server.js#L314)).  So theoretically (...), it should have an additional token added in front of `/packag...` (for example, `##ROOT_URL_PATH_PREFIX##`) which would need to be substituted-in [here in this code](https://github.com/meteor/meteor/blob/devel/packages/oauth/oauth_server.js#L345).

/cc @cunnagin

**Comment by abernix** [comment]
This should be fixed with Meteor 1.4.3.1 thanks to #8081.  Thank you, @koszta!

Thanks, @Buom01, for reporting this originally.  Please report back if you're still having issues after updating to Meteor 1.4.3.1.  You'll also note in the [`History.md`](https://github.com/meteor/meteor/blob/devel/History.md#v1431-2017-02-14) that there are some other changes in the 1.4.3.x-series relating to `accounts-*` packages that should be consulted during the upgrade.

    meteor update --release 1.4.3.1

Closing!
